src/sidebar/reducers/annotations.js

```diff
@@ -146,6 +146,20 @@ var update = {
     return {annotations: []};
   },
 
+  UPDATE_FLAG_STATUS: function (state, action) {
+    var annotations = state.annotations.map(function (annot) {
+      var match = (annot.id && annot.id === action.id);
+      if (match) {
+        return Object.assign({}, annot, {
+          flagged: action.isFlagged,
+        });
+      } else {
+        return annot;
+      }
+    });
+    return {annotations: annotations};
+  },
+
   UPDATE_ANCHOR_STATUS: function (state, action) {
     var annotations = state.annotations.map(function (annot) {
       var match = (annot.id && annot.id === action.id) ||
```

Comment:
The `annot.id` test could actually be removed here since the action must always have an ID and therefore the `===` test can only pass if the annotation also has an id.